backend/app/api/admin_routes/upload.py

```diff
@@ -33,6 +34,15 @@ def upload_files(
                 status_code=status.HTTP_400_BAD_REQUEST,
                 detail="File name cannot be empty",
             )
+        sys_upload_max_body_size = SiteSetting.upload_max_body_size
+        if sys_upload_max_body_size:
+            if file.size > sys_upload_max_body_size:
+                max_body_size_in_mb = sys_upload_max_body_size / 1024 / 1024
+                raise HTTPException(
+                    status_code=status.HTTP_413_REQUEST_ENTITY_TOO_LARGE,
+                    detail=f"File size({file.size} bytes) exceeds maximum allowed size({sys_upload_max_body_size} bytes, {max_body_size_in_mb} MB)",
```

Comment:
To avoid lengthy decimal places:

```python
>> bytes = 12836792 / 1024 / 1024
>> bytes
12.242118835449219
>> print("{:.2f} MiB".format(bytes))
12.24 MiB
```

```suggestion
                    detail="The upload file size ({:.2f} MiB) exceeds maximum allowed size ({:.2f} MiB)".format(upload_file_size_in_mb, max_upload_file_size_in_mb),
```